From: =?utf-8?q?R=C3=A9mi_Palancher?= <remi-externe.palancher@edf.fr>
Date: Fri, 29 Jul 2016 12:23:40 +0200
Subject: Return topology switches/nodes as hostlist

The switches and nodes linked to a switch are now given as a string
hostlist instead of a list splitted by commas that could cause badly
formatted hostlists (eg. node[1-3,4] -> [ 'node[1-3', '4]' ]).

Forwarded: https://github.com/PySlurm/pyslurm/pull/73
---
 pyslurm/pyslurm.pyx | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyslurm/pyslurm.pyx b/pyslurm/pyslurm.pyx
index cdf4cc8..20ee84b 100644
--- a/pyslurm/pyslurm.pyx
+++ b/pyslurm/pyslurm.pyx
@@ -3859,10 +3859,10 @@ cdef class topology:
 
 				name = u"%s" % self._topo_info_ptr.topo_array[i].name
 				Topo_dict[u'name'] = name
-				Topo_dict[u'nodes'] = slurm.listOrNone(self._topo_info_ptr.topo_array[i].nodes, ',')
+				Topo_dict[u'nodes'] = slurm.stringOrNone(self._topo_info_ptr.topo_array[i].nodes, '')
 				Topo_dict[u'level'] = self._topo_info_ptr.topo_array[i].level
 				Topo_dict[u'link_speed'] = self._topo_info_ptr.topo_array[i].link_speed
-				Topo_dict[u'switches'] = slurm.listOrNone(self._topo_info_ptr.topo_array[i].switches, ',')
+				Topo_dict[u'switches'] = slurm.stringOrNone(self._topo_info_ptr.topo_array[i].switches, '')
 
 				Topo[name] = Topo_dict
 
